plugins/lookup/cluster_moid.py

```diff
@@ -0,0 +1,145 @@
+version_added: 1.0.1
+requirements:
+    - vSphere 7.0.2 or greater
+    - python >= 3.6
+    - aiohttp
+'''
+
+
+EXAMPLES = r'''
+# lookup sample
+- name: set connection info
+  set_fact:
+    connection_args:
+        vcenter_hostname: "vcenter.test"
+        vcenter_username: "administrator@vsphere.local"
+        vcenter_password: "1234"
+
+- name: lookup MoID of the object
+  debug: msg="{{ lookup('vmware.vmware_rest.cluster_moid', '/my_dc/host/my_cluster'},
+                        **connection_args) }}"
+
+- name: lookup MoID of the object inside the path
+  debug: msg="{{ lookup('vmware.vmware_rest.cluster_moid', '/my_dc/host/'}) }}"
+'''
+
+
+RETURN = r'''
+_list:
+    description: list of the MoIDs
+    type: list
+    elements: str
+'''
+
+
+import asyncio
+import os
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils._text import to_native
+
+from ansible_collections.cloud.common.plugins.module_utils.turbo.exceptions import EmbeddedModuleFailure
+from ansible_collections.vmware.vmware_rest.plugins.module_utils.vmware_rest import open_session
+from ansible_collections.vmware.vmware_rest.plugins.module_utils.lookup import Lookup
+from ansible_collections.vmware.vmware_rest.plugins.module_utils.lookup import get_credentials
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables, **kwargs):
+        self.set_options(var_options=variables, direct=get_credentials(**kwargs))
+        loop = asyncio.get_event_loop()
+        return loop.run_until_complete(asyncio.gather(self.entry_point(terms)))
+
+    async def entry_point(self, terms):
+        session = None
+        self._options['object_type'] = 'cluster'
+
+        if not self.get_option("vcenter_hostname"):
+            raise AnsibleError("vcenter_hostname cannot be empty")
+        if not self.get_option("vcenter_username"):
+            raise AnsibleError("vcenter_username cannot be empty")
+        if not self.get_option("vcenter_password"):
+            raise AnsibleError("vcenter_password cannot be empty")
+
+        try:
+            session = await open_session(
+                vcenter_hostname=self.get_option("vcenter_hostname"),
+                vcenter_username=self.get_option("vcenter_username"),
+                vcenter_password=self.get_option("vcenter_password"),
+                validate_certs=bool(self.get_option("vcenter_validate_certs")),
+                log_file=self.get_option("vcenter_rest_log_file"),
+            )
+        except EmbeddedModuleFailure as e:
+            raise AnsibleError("Error connecting: %s" % to_native(e))
+
+        self._options["session"] = session
+        lookup = Lookup(self._options)
+
+        if not terms:
+            raise AnsibleError("No object has been specified.")
+
+        tasks = [
+            asyncio.ensure_future(lookup.moid(term)) for term in terms
+        ]
+
+        return [await i for i in tasks]
```

Comment:
A lot of the logic here is duplicated across the different lookup plugin. Can we move it in the lookup.py?

Reply:
I moved this logic inside lookup.py.